readinglist/__init__.py

```diff
@@ -97,9 +108,36 @@ def on_new_response(event):
     config.add_subscriber(on_new_response, NewResponse)
 
 
+def end_of_life_tween_factory(handler, registry):
+    """Pyramid tween to handle service end of life."""
+
+    def eos_tween(request):
+        eos_date = registry.settings.get("readinglist.eos")
+        eos_url = registry.settings.get("readinglist.eos_url")
+        if eos_date:
+            eos_date = dateparser.parse(eos_date)
```

Comment:
* Creating a new app with different settings would mean creating two different apps because I'm testing with a date in the past and in the future (and with no date)
* Unit testing wouldn't be enough here since I found some problems only when having it enabled within the scope of the app.
* I don't get what the cache proposal you mention is (it seems that if we cache in the closure we have the same problem, but I may miss something)